juriscraper/opinions/united_states/state/mich.py

```diff
@@ -3,132 +3,75 @@
 Court Short Name: Mich.
 Contact: sitefeedback@courts.mi.gov
 History:
- - 2014-09-21: Updated by Jon Andersen to handle some fields being missing
- - 2014-08-05: Updated to have a dynamic URL, an oversight during check in.
+    - 2014-09-21: Updated by Jon Andersen to handle some fields being missing
+    - 2014-08-05: Updated to have a dynamic URL, an oversight during check in.
+    - 2022-01-28: Updated for new web site, @satsuki-chan.
 """
-
+import re
 import time
-from datetime import date, timedelta
+from typing import List
 
-from juriscraper.lib.string_utils import clean_if_py3, titlecase
-from juriscraper.OpinionSite import OpinionSite
+from juriscraper.AbstractSite import logger
+from juriscraper.lib.string_utils import titlecase
+from juriscraper.OpinionSiteLinear import OpinionSiteLinear
 
 
-class Site(OpinionSite):
+class Site(OpinionSiteLinear):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.today = date.today()
-        self.a_while_ago = date.today() - timedelta(days=30)
-        self.url = (
-            "http://courts.mi.gov/opinions_orders/opinions_orders/"
-            "Pages/default.aspx?SearchType=4"
-            "&Status_Advanced=sct&FirstDate_Advanced="
-            "{start_month}%2f{start_day}%2f{start_year}"
-            "&LastDate_Advanced="
-            "{end_month}%2f{end_day}%2f{end_year}".format(
-                start_day=self.a_while_ago.day,
-                start_month=self.a_while_ago.month,
-                start_year=self.a_while_ago.year,
-                end_day=self.today.day,
-                end_month=self.today.month,
-                end_year=self.today.year,
-            )
-        )
-        self.back_scrape_iterable = list(range(0, 868))
         self.court_id = self.__module__
+        self.url = "https://www.courts.michigan.gov/api/CaseSearch/SearchCaseOpinions?resource=supreme-court-opinion&aAppellateCourt=Supreme%20Court&searchQuery=&sortOrder=Newest&page=1&pageSize=100"
+        self.back_scrape_iterable = list(range(1, 99))
+        self.request["headers"] = {"User-Agent": ""}
+        self.title_re = (
+            r"(MSC|COA) (?P<docket>\d+)\s+(?P<name>.+)\s+(Opinion|Order)"
```

Comment:
Let's not merge court of appeals into the Supreme Court scraper.

Reply:
Regular expression for both courts separated